doc/examples/build_matrix.ml

```diff
@@ -14,9 +22,9 @@ let dockerfile ~base ~ocaml_version =
   workdir "/src" @@
   add ~src:["*.opam"] ~dst:"/src/" () @@
   env ["OPAMERRLOGLEN", "0"] @@
-  run "opam install . --show-actions --deps-only -t | awk '/- install/{print $3}' | xargs opam depext -iy" @@
+  run "opam-2.1 install . --show-actions --deps-only -t" @@
   copy ~src:["."] ~dst:"/src/" () @@
-  run "opam install -tv ."
+  run "opam-2.1 install -tv ."
```

Comment:
```suggestion
  run "ln -f /usr/bin/opam-2.1 /usr/bin/opam" @@
  run "opam init --reinit -n" @@
  add ~src:["*.opam"] ~dst:"/src/" () @@
  env ["OPAMERRLOGLEN", "0"] @@
  run "opam install . --show-actions --deps-only -t" @@
```
opam will upgrade the config and local state the first time it is run, however this takes time and is not backward compatible so it’s better to separate the upgrade from the rest to have a predictible base and to avoid issues with package that call opam itself (which would fail)

Reply:
Added a sudo for the `ln -f ` line but otherwise the same as suggested. Thanks @kit-ty-kate